### PR TITLE
VACMS-6071: Create three new VAMC Top Task content types.

### PR DESCRIPTION
**Pull request by ethanteague**
## Description

Relates to #6071
## Testing done
Visual, phpunit, Behat

## Screenshots
image
image
image

## QA steps
- As an administrator (or content_admin), create 1 node here: `/node/add/vamc_system_medical_records_offi` & here `/node/add/vamc_system_register_for_care` & here: `/node/add/vamc_system_billing_insurance`
- Prior to save, visually verify the pages match the screenshots above
- After save, visually verify that title is `Register for care` or `Medical records office` or `Billing and insurance`
- Login as ryan.thurlwell@va.gov and confirm that you cannot view any of the three new node add forms
- Visit all three of the new nodes created by admin, and confirm that you can edit and save each of the new nodes.

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Core Application Team`
- [x] `Product Support Team`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change? 
- [ ] Yes, and it's written in issue ____ and queued for publication. 
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written 
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change. 
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

File: config/sync/metatag.metatag_defaults.node__vamc_system_billing_insurance.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+uuid: 360ab26e-baf7-4613-a557-eaca6f52c7e3
+langcode: en
+status: true
+dependencies: {  }
+id: node__vamc_system_billing_insurance
+label: 'Content: VAMC System Billing Insurance'
```
Comment (kevwalsh) [inline config/sync/metatag.metatag_defaults.node__vamc_system_billing_insurance.yml:6]: should be "VAMC System Billing and Insurance"

File: config/sync/node.type.vamc_system_billing_insurance.yml [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+      - va-wichita-health-care
+      - va-wilkes-barre-health-care
+      - va-wilmington-health-care
+    parent: 'main:'
+  node_title_help_text:
+    title_help: ''
+  menu_force:
+    menu_force: 1
+    menu_force_parent: 0
+name: 'VAMC System Billing Insurance'
```
Comment (kevwalsh) [inline config/sync/node.type.vamc_system_billing_insurance.yml:160]: should be "VAMC System Billing and Insurance"

File: tests/behat/drupal-spec-tool/content_model_bundles.feature [hunk 3]
```diff
@@ -46,6 +46,9 @@ Feature: Content model bundles
 | VAMC System Health Service | regional_health_care_service_des | Content type | A description of a health service specific to a VAMC system, which appears on a VAMC's health services page and on facility pages, within accordions. |
 | VAMC System Operating Status | vamc_operating_status_and_alerts | Content type | Create one of these pages for each VAMC system. Then you can add banner alerts and update facilities' operating status, all from one place. |
 | VAMC System Policies Page | vamc_system_policies_page | Content type | Add policies specific to this VA medical center to appear on the Policies page. Local policies will appear alongside national policies that apply to all VAMCs. |
+| VAMC System Register for Care | vamc_system_register_for_care | Content type |  |
+| VAMC System Medical Records Office | vamc_system_medical_records_offi | Content type |  |
+| VAMC System Billing Insurance | vamc_system_billing_insurance | Content type |  |
```
Comment (kevwalsh) [inline tests/behat/drupal-spec-tool/content_model_bundles.feature:51]: should be "VAMC System Billing and Insurance"

File: config/sync/metatag.metatag_defaults.node__vamc_system_billing_insurance.yml (hunk 1 above)
Comment (kevwalsh) [inline config/sync/metatag.metatag_defaults.node__vamc_system_billing_insurance.yml:6]: should be "VAMC System Billing and Insurance"

**Review by kevwalsh — CHANGES_REQUESTED** [review]
- [ ] "Billing and insurance" should have the "and" everywhere except the machine name, where it doesn't fit.  
- [ ] Section should always appear first, and in these cases above Menu settings. i think the only way to do this is to set the section fieldset's weight manually. 

Create_VAMC_System_Billing_Insurance___VA_gov_CMS_and_VACMS-6071_create_three_new_content_types_by_ethanteague_·_Pull_Request__6142_·_department-of-veterans-affairs_va_gov-cms

**Comment by ethanteague** [comment]
Updated, @kevwalsh :+1:

**Comment by swirtSJW** [comment]
Can we get the system to be sorted by name?

**Comment by swirtSJW** [comment]
image
I don't think this is what we want.   This would treat unique on the combination of the two.  I think we want two allow only ones,  one on section, one on system.

**Comment by ethanteague** [comment]
> image
> I don't think this is what we want. This would treat unique on the combination of the two. I think we want two allow only ones, one on section, one on system.



> Can we get the system to be sorted by name?
> 

I think this would require creating a new view entity reference view display (the field / form settings don't have any sorting options / magic available). Happy to do it, but will mean adding config, and will result in having other instances of this field having different config until we update to use the new view reference.

Just tell me how you all want me to proceed @kevwalsh @swirtSJW

**Comment by kevwalsh** [comment]
You can sort the select list with a setting in the field (not on the form widget). It's confusing because it doesn't say "Title". you can see how this is done in some of the other field_office select list settings. 

and yes, only one of each content type per section. and only one of each content type per field_office

**Comment by ethanteague** [comment]
Updated, @swirtSJW

File: config/sync/migrate_plus.migration.va_node_health_care_local_facility_status.yml [hunk 4]
```diff
@@ -57,7 +57,6 @@ source:
     - 'https://www.dublin.va.gov/locations/facilities.asp'
     - 'https://www.durham.va.gov/locations/facilities.asp'
     - 'https://www.elpaso.va.gov/locations/facilities.asp'
-    - 'https://www.erie.va.gov/locations/facilities.asp'
```
Comment (swirtSJW) [inline config/sync/migrate_plus.migration.va_node_health_care_local_facility_status.yml:60]: I don't think this belongs here.

File: docroot/modules/custom/va_gov_migrate/config/install/migrate_plus.migration.va_node_health_care_local_facility_status.yml [hunk 5]
```diff
@@ -58,7 +58,6 @@ source:
     - https://www.dublin.va.gov/locations/facilities.asp
     - https://www.durham.va.gov/locations/facilities.asp
     - https://www.elpaso.va.gov/locations/facilities.asp
-    - https://www.erie.va.gov/locations/facilities.asp
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_migrate/config/install/migrate_plus.migration.va_node_health_care_local_facility_status.yml:61]: Does not belong here.

**Comment by ethanteague** [comment]
updated, @swirtSJW :+1:

**Review by swirtSJW — APPROVED** [review]
There is an issue where the node title is automatically set, but it is not auto setting the menu title but I will create a separate issue for this to move this one along.

**Pull request merged**